src/devices/src/virtio/block/persist.rs

```diff
@@ -19,17 +20,63 @@ use super::*;
 use crate::virtio::persist::VirtioDeviceState;
 use crate::virtio::{DeviceState, TYPE_BLOCK};
 
+#[derive(Clone, Copy, Versionize, PartialEq)]
+// NOTICE: Any changes to this structure require a snapshot version bump.
+pub enum CacheTypeState {
+    Unsafe,
+    Writeback,
+}
```

Comment:
nitish: this being a simple enum, we could have just used the original instead of creating a shadow _state_ one.

Reply:
My understanding is that we should not include any `persist` logic in our devices, so it didn't make sense to add a `#[derive(Versionize)]` on `CacheType`.

Furthermore, `CacheType` is a part of the block device and has little to do with any snapshot logic, so it didn't make sense to have it in `block/persist.rs` either.

This was the reasoning behind creating a shadow struct for `CacheType`.

Reply:
> My understanding is that we should not include any persist logic in our devices

I don't think that's accurate. Plain-old-data structs can be versioned directly I think. @sandreim can confirm/disprove this, I might be missing something.

nit: In current PR `CacheType` already has versionize.

Reply:
My take on this is to keep our devices clean of any serialization logic. This is mostly forward looking towards ingesting upstream rust-vmm devices more easily.